packages/@aws-cdk/aws-lambda-python/lib/bundling.ts

```diff
@@ -16,39 +26,65 @@ export interface BundlingOptions {
    * The runtime of the lambda function
    */
   readonly runtime: lambda.Runtime;
+
+  /**
+   * Output path suffix ('python' for a layer, '.' otherwise)
+   */
+  readonly outputPathSuffix: string;
 }
 
 /**
  * Produce bundled Lambda asset code
  */
 export function bundle(options: BundlingOptions): lambda.AssetCode {
-  // Bundling image derived from runtime bundling image (AWS SAM docker image)
-  const image = cdk.BundlingDockerImage.fromAsset(__dirname, {
-    buildArgs: {
-      IMAGE: options.runtime.bundlingDockerImage.image,
-    },
-  });
-
-  let installer = options.runtime === lambda.Runtime.PYTHON_2_7 ? Installer.PIP : Installer.PIP3;
+  const { entry, runtime, outputPathSuffix: pathSuffix } = options;
 
-  let hasRequirements = fs.existsSync(path.join(options.entry, 'requirements.txt'));
+  const hasDeps = hasDependencies(entry);
 
-  let depsCommand = chain([
-    hasRequirements ? `${installer} install -r requirements.txt -t ${cdk.AssetStaging.BUNDLING_OUTPUT_DIR}` : '',
-    `rsync -r . ${cdk.AssetStaging.BUNDLING_OUTPUT_DIR}`,
+  const depsCommand = chain([
+    hasDeps ? `rsync -r ${BUNDLER_DEPENDENCIES_CACHE}/. ${cdk.AssetStaging.BUNDLING_OUTPUT_DIR}/${pathSuffix}` : '',
+    `rsync -r . ${cdk.AssetStaging.BUNDLING_OUTPUT_DIR}/${pathSuffix}`,
   ]);
```

Comment:
```suggestion
  const depsCommands = [
    `rsync -r . ${cdk.AssetStaging.BUNDLING_OUTPUT_DIR}/${pathSuffix}`,
  ]
  
  if (hasDeps) {
    depsCommands.unshift(`rsync -r ${BUNDLER_DEPENDENCIES_CACHE}/. ${cdk.AssetStaging.BUNDLING_OUTPUT_DIR}/${pathSuffix}`);
  }

  const depsCommand = chain(depsCommands);
```

Reply:
Technically order of command execution doesn't matter here, right? In which case `.push` might be just slightly easier to follow.